### PR TITLE
Add border visual tests

### PR DESCRIPTION
**Pull request by BretJohnson**
### Description of Change

Fully enable the border tests, started as part of https://github.com/dotnet/maui/pull/17310.
We decided to break the tests out of 17310, into this PR, so we can merge 17310 ASAP and merge the test PR (this one) shortly after.

So the sequence should be:
- ~~Finalize https://github.com/dotnet/maui/pull/17310 and merge it (hopefully very shortly today).~~
- ~~Rebase this PR on latest main.~~
- Finalize and merge this PR, adding the UI tests (hopefully later today - it's close, but we don't want to block on it).

### CODE REVIEW

**Review by jstedfast — DISMISSED** [review]

**Review by jstedfast — DISMISSED** [review]

**Review by jsuarezruiz — COMMENTED** [review]
image
`System.TimeoutException : Timed out waiting for element...`
Launching the test processes on CI again to see if can pass the added ones.

File: src/Controls/tests/UITests/snapshots/android/BordersWithVariousShapes.png
Comment (jstedfast) [inline src/Controls/tests/UITests/snapshots/android/BordersWithVariousShapes.png:1]: why did the height of this image shrink? Seems like it's not the correct proportions?

File: src/Controls/tests/UITests/snapshots/android/BorderBackground.png
Comment (jstedfast) [inline src/Controls/tests/UITests/snapshots/android/BorderBackground.png:1]: Either the Android platform is broken for this test or iOS is :-(

File: src/Controls/tests/UITests/snapshots/android/BordersWithVariousShapes.png
Comment (BretJohnson) [reply src/Controls/tests/UITests/snapshots/android/BordersWithVariousShapes.png:1]: The Android image I had previously added form the Android emulator on my machine - and apparently I didn't use the right emulator image. I updated to use the image from the emulator used CI, which has a different screen size.

File: src/Controls/tests/UITests/snapshots/android/BorderBackground.png
Comment (BretJohnson) [reply src/Controls/tests/UITests/snapshots/android/BorderBackground.png:1]: > Either the Android platform is broken for this test or iOS is :-(

Android looks like it matches Windows. So iOS may be wrong.

File: src/Controls/tests/UITests/snapshots/android/BordersWithVariousShapes.png
Comment (jstedfast) [reply src/Controls/tests/UITests/snapshots/android/BordersWithVariousShapes.png:1]: oh, weird, but ok.

**Review by jstedfast — APPROVED** [review]

**Pull request closed without merging**